collectors/proc.plugin/proc_net_dev.c

```diff
@@ -659,6 +693,39 @@ int do_proc_net_dev(int update_every, usec_t dt) {
             d->tcarrier    = str2kernel_uint_t(procfile_lineword(ff, l, 15));
         }
 
+        if (d->do_duplex != CONFIG_BOOLEAN_NO && d->filename_duplex) {
+            char buffer[32 + 1];
+
+            if (read_file(d->filename_duplex, buffer, 32)) {
+                error("Cannot refresh interface %s duplex state by reading '%s'. I will stop updating it.", d->name, d->filename_duplex);
+                freez(d->filename_duplex);
+                d->filename_duplex = NULL;
+            } else {
+                // values can be unknown, half or full -- just check the first letter for speed
+                if (buffer[0] == 'f')
+                    d->duplex = 2;
+                else if (buffer[0] == 'h')
+                    d->duplex = 1;
+                else
+                    d->duplex = 0;
+            }
+        }
+
+        if(d->do_operstate != CONFIG_BOOLEAN_NO && d->filename_operstate) {
+            char buffer[32 + 1], *trimmed_buffer;
```

Comment:
If you accepts the previous suggestion, please, add it here too. Why do you have the "same variable" declared in different scopes? I think we could use only one outside these conditions. Did you receive any warning to reduce scopes?

Reply:
I didn't get any warnings but I don't see the point of making the scope of the variable too wide (half of the whole code of the `proc/net` module).